### PR TITLE
feat: Updated the typescript scaffold

### PR DESCRIPTION
**Pull request by adilansari**
- The scaffolding includes complete implementations for - `tigrisClient.ts`, `models/user.ts`
- And a boilerplate code for `repository/users.ts` for user to implement using guide, 
- And empty `testScript.ts` that user will intialize client and write queries against collection class using guide
- Once implemented, user is expected to execute `testScript.ts` via `npm run test`

### CODE REVIEW

File: bin/index.js [hunk 1]
```diff
@@ -1,288 +1,220 @@
...
+  if (!fs.existsSync(srcDir)) {
+    fs.mkdirSync(srcDir)
+  }
+
+  const modelsDir = outputDir + '/src/models'
+  if (!fs.existsSync(modelsDir)) {
+    fs.mkdirSync(modelsDir)
+  }
+
+  const collectionsDir = outputDir + '/src/collections'
```
Comment (ovaistariq) [inline bin/index.js:75]: Is the convention to have a collections directory? Or would the code go somewhere else.

File: bin/index.js [hunk 2]
```diff
@@ -1,288 +1,220 @@
...
-        '    },\n' +
-        '    name: {\n' +
-        '        type: TigrisDataTypes.STRING,\n' +
-        '    },\n' +
-        '    balance: {\n' +
-        '        type: TigrisDataTypes.NUMBER,\n' +
-        '    },\n' +
-        '};\n'
+function initializeModels (outputDir) {
+  const content = 'import {\n' +
```
Comment (ovaistariq) [inline bin/index.js:146]: Is it easier to have the content in a file that you then read here.

File: bin/index.js (hunk 1 above)
Comment (adilansari) [reply bin/index.js:75]: probably repository

File: bin/index.js (hunk 2 above)
Comment (adilansari) [reply bin/index.js:146]: This seems a little complicated for now, will need more investment

**Review by JigarJoshi — APPROVED** [review]

**Pull request merged**